src/agent/agent.c

```diff
@@ -1813,6 +1813,17 @@ static bool ensure_orch_address(Agent *agent) {
         }
 
 
+        if (!isIPv4Addr(agent->host) && !isIPv6Addr(agent->host)) {
+                char *ip_address = malloc0_array(0, sizeof(char), INET6_ADDRSTRLEN);
+                int r = get_address(agent->host, &ip_address);
+                if (r != 0) {
+                        hirte_log_errorf("Failed to get IP address from: '%s'", agent->host);
+                        return false;
+                }
+                hirte_log_infof("Translated '%s' to '%s'", agent->host, ip_address);
+                copy_str(&agent->host, ip_address);
```

Comment:
Hmm, aren't we loosing the original hostname in the current agent instance? Wouldn't it be safer to keep the original hostname and do a resolve on each reconnection?

Reply:
Good point! I am not sure if the resolved IP would change, but it feels off to change the configured host. 

I think the `ensure_orch_address` in the `agent_start` 
https://github.com/containers/hirte/blob/main/src/agent/agent.c#L1866-L1868
should be copied to
https://github.com/containers/hirte/blob/main/src/agent/agent.c#L2058-L2059
Where `agent->orch_addr` is set to `NULL` before. But its executed only when the `agent->connection_state == AGENT_CONNECTION_STATE_RETRY` to avoid double resolving on startup (since the early first `ensure_orch_address` in the agent startup ensures a fail fast, fail early)

Reply:
> Hmm, aren't we loosing the original hostname in the current agent instance? Wouldn't it be safer to keep the original
> hostname and do a resolve on each reconnection?

We don't use the hostname from manager in the code (other than resolve).  We could create a new field in the struct and keep it in `agent->manager_fqdn` when set, otherwise it will be NULL. However, this could be a different patch/issue.

Reply:
Yes, lets do that in a followup PR.